docker/docker.go

```diff
@@ -150,8 +154,41 @@ func localImageExists(ctx context.Context, dClient *client.Client, hostname stri
 	return true, nil
 }
 
-func pullImage(ctx context.Context, dClient *client.Client, hostname string, containerImage string) error {
-	out, err := dClient.ImagePull(ctx, containerImage, types.ImagePullOptions{})
+func pullImage(ctx context.Context, dClient *client.Client, hostname string, containerImage string, prs []v3.PrivateRegistry) error {
+
+	pullOptions := types.ImagePullOptions{}
+	prsMap := make(map[string]v3.PrivateRegistry)
+	// build the Private Registry map
+	if len(prs) > 0 {
```

Comment:
@moelsayed do you think it would make sense to pass the map all way through? Right from the very top. So this way you won't have to rebuild it from list every time pullImage is called

Reply:
There are a log of places where this call would be triggered. I would  need to rebuild it at every to level call or add it to the cluster option to use it later. I will look into the second option.